tree.go

```diff
@@ -552,24 +570,105 @@ func (n *InternalNode) Commitment() *Point {
 }
 
 func (n *InternalNode) Commit() *Point {
-	poly := make([]Fr, NodeWidth)
-	emptyChildren := 256
-
+	if n.depth <= 1 {
+		return n.commitRoot()
+	}
 	if len(n.cow) != 0 {
+		polyp := frPool.Get().(*[]Fr)
+		poly := *polyp
+		defer func() {
+			for i := 0; i < NodeWidth; i++ {
+				poly[i] = Fr{}
+			}
+			frPool.Put(polyp)
+		}()
```

Comment:
We use a `frpool` to avoid so many allocations of 256 `[]Fr`.
A `[256]Fr` is a big chunk of memory. A `Fr` is a `[4]uint64`, so `[256]Fr` is 8KiB of memory, which is a lot. 

This puts a lot of pressure on the GC, which not only eats a lot of memory but also a lot of CPU since the GC has to collect all this garbage. Saving this memory improved the wall-clock time of the go-ethereum benchmark.